CRM/Contact/CustomSearchTask.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ * Class to represent the actions that can be performed on a group of contacts used by the search forms.
+ */
+class CRM_Contact_CustomSearchTask extends CRM_Core_Task {
+
+  /**
+   * @var string
+   */
+  public static $objectType = 'contact';
+
+  public static function tasks() {
+    if (!self::$_tasks) {
+      parent::tasks();
+    }
+    return self::$_tasks;
+  }
```

Comment:
I can't see how this function is needed - all it does is call the parent?

Reply:
It does check whether `self::$_tasks` is set. If not call parent. If it does return self::$_task. So it does do more than mere calling the parent.

Reply:
So `self::$tasks` is meant to be an integration point? E.g. a custom search class can set that variable to override the defaults? How would that work?